File: java/spring-mongo/src/main/java/com/bmarques/springmongo/invoice/InvoiceService.java
package com.bmarques.springmongo.invoice;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class InvoiceService {

    @Autowired
    private InvoiceRepository repository;

    public Invoice save(Invoice invoice) {
        return repository.save(invoice);
    }

    public Invoice findById(Integer id) {
        return repository.findById(id).orElseThrow(() -> new InvoiceNotFoundException("Invoice not found"));
    }

    public List<Invoice> findAll() {
        return repository.findAll();
    }

    public void deleteById(Integer id) {
        repository.deleteById(id);
    }
}

File: kotlin/spring-mongo/src/main/kotlin/com/bmarques/springmongo/invoice/InvoiceService.kt
package com.bmarques.springmongo.invoice

import org.springframework.stereotype.Service

@Service
class InvoiceService(private val repository: InvoiceRepository) {
    fun save(invoice: Invoice): Invoice = repository.save(invoice)

    fun findById(id: Int): Invoice = repository.findById(id).orElseThrow { InvoiceNotFoundException("ID $id not found") }

    fun findAll(): List<Invoice> = repository.findAll()

    fun deleteById(id: Int) = repository.deleteById(id)
}
